services/api-server/src/simcore_service_api_server/utils/client_decorators.py

```diff
@@ -0,0 +1,92 @@
+""" Collection of decorators for httpx request functions
+
+    Each decorator implements a specific feature on the request workflow:
+    - retrial
+    - error handling
+    - transform/compose/merge responses from different internal APIs
+    - TODO: circuit breaker?
+    - TODO: diagnostic tracker?
+    - TODO: cache?
+"""
+
+import functools
```

Comment:
shall we consider having a base library for the fastapi-based services? this could become useful so that we have similar behaviours between all the fastapi servers.

Reply:
yes